tests/Makefile

```diff
@@ -7,6 +7,9 @@ SHIM_FILES=${SRC_PATH}/lib/*.cpp
 PSC_FILE=../src/PubSubClient.cpp
 CC=g++
 CFLAGS=-std=c++11 -I${SRC_PATH}/lib -I../src
+CFLAGS+=-pedantic -Werror -Wall -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Wdisabled-optimization -Wformat=2
+CFLAGS+=-Winit-self -Wmissing-declarations -Wmissing-include-dirs -Woverloaded-virtual -Wredundant-decls
+CFLAGS+=-Wsign-conversion -Wsign-promo -Wstrict-overflow=5 -Wswitch-default -Wundef -Wno-unused
```

Comment:
Maybe also `-Wnull-dereference`
And `-Wno-return` (not 100% sure about the actual warning name)

Reply:
I added the -Wnull-dereference. -Wno-return is unknown. What warning do you mean?
https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

Reply:
I couldn't find it in my project anymore, so I guess it may now be set default for ESP32/Arduino?

Not sure if it is the `-Wmissing-noreturn`.

Anyway it should trigger on these kinds of functions:
```c++
bool foo() {
  // your code...
  // No return
}
```
At least on ESP32, this would result in an immediate crash. (as it should...)

Reply:
Yes, we get these kind of warnings / errors ...
```
bool PubSubClient::tst(const char* string) {

}
```

```
../src/PubSubClient.cpp:615:36: error: unused parameter 'string' [-Werror,-Wunused-parameter]
  615 | bool PubSubClient::tst(const char* string) {
      |                                    ^
../src/PubSubClient.cpp:617:1: error: non-void function does not return a value [-Werror,-Wreturn-type]
  617 | }
      | ^
2 errors generated.
```
It's `-Wreturn-type` ...